### DISCUSSION TITLE
redis-cluster support

### DISCUSSION THREAD

**Discussion by pat-s** [authelia/authelia#5474] [Ideas]
Hi there!

I've read that redis and redis-sentinel are currently supported. Is there any ETA for redis-cluster? We're running the latter and would need to make use of it for Authelia. We don't want to deploy a separate redis instance just for this, especially because redis-cluster is already HA. 

In many other applications switching the connection string to `redis+cluster` is often enough to tell `go-redis` that it's communicating with a redis-cluster instance. Not sure if this would also be true for authelia?

**Comment by james-d-elliott** [comment]
We generally don't give specific ETA's. Features are based usually on community interest. That being said there is a rework on the table for sessions which would include a change to redis/redis sentinel which may have scope for adding cluster.

**Comment by archef2000** [comment]
Would like to connect authelia to my redis cluster but I just get the moved and 'Unable to save updated user session' errors. @james-d-elliott Is it already implemented and I am just doing something wrong or is it still in dev?

**Reply by Crowley723** [reply]
The mentioned session rework, to my knowledge, hasn't been started yet. There are still things that have to be done for v4.39.0. I think it's likely we may see the session changes in v4.40, but as James mentioned, we generally try to avoid ETAs on features.

**Reply by archef2000** [reply]
Sorry for my ignorance but the libary you are using for redis conndctions does support redis clusters so it would just need to be enabled via the config to follow the redirects. right?

**Reply by james-d-elliott** [reply]
Technically it does however making a fairly sizable change before refactoring is just not going to feel good for users. The other session related features are widely requested and redis cluster support has very low interest, so it's unfortunately a measure of benefits.
